calicoctl/calico_ctl/node.py

```diff
@@ -847,8 +848,13 @@ def handle_sigterm(sig, frame):
     except KeyboardInterrupt:
         # Mainline. Someone pressed Ctrl-C.
         print "Stopping Calico node..."
+        clean_exit = True
     finally:
         # Could either be this process is being killed, or output generator
         # raises an exception.
         if stop_container_on_exit:
             docker_client.stop(container)
+            # If the container is stopped due to unknown circumstances, exit
+            # with failed status code.
+            if not clean_exit:
+                sys.exit(1)
```

Comment:
Actually you could do what dan suggests:

```
try:
    rc = docker_client.wait(container)
    sys.exit(rc)
except KeyboardException:
    sys.exit(0)
finally:
    do finally stuff
```

This should work IIRC because sys.exit() just raises a SystemExit exception, so you're finally branch should still be hit.

Couple of things though:  (1) we need to make it super clear that this method will always do a sys.exit() - although you could arguably remove the sys.exit() from the Keyboardinterrupt branch and let it percolate up.

and (2) do we lose any useful logging if we use wait rather than attach?